src/utilities/viewer-settings.cpp

```diff
@@ -0,0 +1,74 @@
+/*
+ * OpenKneeboard
+ *
+ * Copyright (C) 2022 Fred Emmott <fred@fredemmott.com>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; version 2.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301,
+ * USA.
+ */
+#include "viewer-settings.h"
+
+#include <OpenKneeboard/Filesystem.h>
+
+#include <OpenKneeboard/json.h>
+#include <OpenKneeboard/utf8.h>
+
+#include <algorithm>
+#include <format>
+#include <fstream>
+
+namespace OpenKneeboard {
+ViewerSettings ViewerSettings::Load() {
+  ViewerSettings ret;
+
+  const auto path = Filesystem::GetSettingsDirectory() / "Viewer.json";
+  if (std::filesystem::exists(path)) {
+    std::ifstream f(path.c_str());
+    nlohmann::json json;
+    f >> json;
+    ret = json;
+  }
+
+  return ret;
+}
+
+void ViewerSettings::Save() {
+  const auto dir = Filesystem::GetSettingsDirectory();
+  if (!std::filesystem::exists(dir)) {
+    std::filesystem::create_directories(dir);
+  }
+  nlohmann::json j = *this;
```

Comment:
For better handling of additional settings in the future, or changing defaults:

```C++
nlohmann::json j;
if (std::filesystem::exists(path)) {
  // ... load from file to j
}
to_json_with_default(j, ViewerSettings {}, *this);
```

With this, the SPARSE macros will:
- save values that are manually changed
- not save default values, unless they were previously changed from the defaults
- properly handle any missing values